src/core/include/units/math.h

```diff
@@ -219,4 +219,45 @@ template<Unit To, typename D, typename U, typename Rep>
   }
 }
 
+/**
+ * @brief Computes the nearest quantity with integer representation and unit type To to q
+ *
+ * Rounding halfway cases away from zero, regardless of the current rounding mode.
+ *
+ * @tparam q Quantity being the base of the operation
+ * @return Quantity The rounded quantity with unit type To
+ */
+template<Unit To, typename D, typename U, typename Rep>
+[[nodiscard]] constexpr quantity<D, To, Rep> round(const quantity<D, U, Rep>& q) noexcept
+  requires (
+    (!treat_as_floating_point<Rep>) ||
+    requires { round(q.number()); } ||
+    requires { std::round(q.number()); }
+  ) // && requires { units::floor(q); }
+{
+  if constexpr(std::is_same_v<To, U>) {
+    if constexpr(treat_as_floating_point<Rep>) {
+      using std::round;
+      return quantity<D, To, Rep>(round(q.number()));
+    }
+    else {
+      return q;
+    }
+  }
+  else {
+    const auto res_low = units::floor<To>(q);
+    const auto res_high = res_low + decltype(res_low)::one();
+    const auto diff0 = q - res_low;
+    const auto diff1 = res_high - q;
+    if (diff0 == diff1) {
+      if (static_cast<int>(res_low.number()) & 1)
```

Comment:
Well, this is tricky. Probably `::one()` should be used here and the user should be responsible for an operator for a custom type if needed.

Reply:
But even if we use `::one` instead of `1`, we still have the problem of using the bitwise and with floating point types